tests/PHPStan/Analyser/data/array-filter.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace ArrayFilter;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @param mixed[] $list
+ */
+function callableString1(array $list): void
+{
+	$filtered = array_filter($list, 'is_string');
+	assertType('array<string>', $filtered);
+}
+
+/**
+ * @param int[] $list
+ */
+function callableString2(array $list): void
+{
+	$filtered = array_filter($list, 'is_string');
+	assertType('array<*NEVER*>', $filtered);
```

Comment:
Hmm is it legal to have a array of never?
Never actually means execution would stop, and I think thats not what going to happen

Reply:
Yeah, that's a strange one. I would have expected an empty array (because the array consists out of ints). Interestingly the same result happens with the following
```php
/**
 * @param int[] $list
 */
function closureTest(array $list): void
{
	$filtered = array_filter($list, fn(mixed $item) => is_string($item));
	assertType('array<*NEVER*>', $filtered);
}
```
So it must be coming from somewhere else at least.

Reply:
You might found a unrelated bug :)

Reply:
Could be. Or because `$scope->filterByTruthyValue($expr);` is used to determine the type with the `is_string` expression and maybe this is expected. But it still is confusing and if this is not a bug I guess it could be handled in the extension somehow.

Reply:
I'll deal with that in a separate PR and adapt this one afterwards -> https://github.com/phpstan/phpstan-src/pull/939